### ISSUE TITLE
Move hakyll website build to stack instead of cabal

### ISSUE THREAD

**Issue by sakshamsharma** [jaspervdj/hakyll#431]
I've personally had lots of trouble with cabal dependencies not resolving very often. Often leads to broken CIs, like the current one for [hakyll itself](https://travis-ci.org/jaspervdj/hakyll)
I recently shifted my new website to use stack instead, which atleast is guaranteed to work most of the times. Perhaps a similar shift for the hakyll website would help?
Here's a link to my travis setup: https://github.com/sakshamsharma/acehack/blob/master/.travis.yml
Although yes, I'm not building hakyll itself every time. That can be changed.
Perhaps I can PR later if this seems like a good idea.

**Comment by jaspervdj** [comment]
I'd be happy to merge such a PR. I haven't been able to spend any time on fixing the broken Travis build for Hakyll, so it's definitely welcome.

**Comment by sakshamsharma** [comment]
Can you detail the required features in that case?
Its just supposed to build the latest code and crash if there was an error? No push/deploys etc for the website? That should not be too difficult :smile:
I'll try to PR in a day or two.

**Comment by jaspervdj** [comment]
It should just run `stack test`. Additionally, you can check if we are able to compile the Hakyll website itself. It should not handle deploys. The website is hosted on my private server, the deploying method is manual right now. :-)

**Comment by sakshamsharma** [comment]
Perhaps unrelated, but I recently hosted a website on a private server with a CI too, using github hooks on pushes to a particular branch. The hook would run a secret script which would pull the latest built files onto the private server. Perhaps not needed for the Hakyll website, since it doesn't get updated too often I believe.
I documented it for Jekyll [here](https://pclub.in/pclub/2016/05/13/pclubci.html), but should be almost the same for Hakyll. Just in case you're interested.
P.S. Hakyll is awesome, thanks! :smile:
